bundles/binding/org.openhab.binding.sallegra/src/main/java/org/openhab/binding/sallegra/internal/SallegraBinding.java

```diff
@@ -0,0 +1,353 @@
+	                        value = node.getRelay(bindingConfig.getCmdValue());
+	                        break;
+	                    case DIMMER:
+	                        value = node.getDimmer(bindingConfig.getCmdValue());
+	                        break;
+	                    case INPUT:
+	                        value = node.getInput(bindingConfig.getCmdValue());
+	                        break; 
+	                    default:
+	                        break;
+                    }
+                    //If a value is got returned update it in openhab
+                    if (value != null) {
+                        postUpdate(provider, bindingConfig.getItem(), value);
+                    }
+                } else {
+                    logger.error("Unknown deviceId \"{}\"", deviceId);
+                }
+            }
+        }
+		
+	}
+
+	
+	
+    private void postUpdate(SallegraBindingProvider provider, Item item,
+            final String value) {
+        Class<? extends Item> itemType = provider.getItemType(item.getName());
+        State state = createState(itemType, value);
+
+        if (state != null) {
+            if (item instanceof GenericItem) {
+                ((GenericItem) item).setState(state);
+            }
+            eventPublisher.postUpdate(item.getName(), state);
+        }
+    }   
+
+    private State createState(Class<? extends Item> itemType,
+            String transformedResponse) {
+        try {
+            if (itemType.isAssignableFrom(NumberItem.class)) {
+                return DecimalType.valueOf(transformedResponse);
+            } else if (itemType.isAssignableFrom(SwitchItem.class)) {
+                return OnOffType.valueOf(transformedResponse);
+            } else if (itemType.isAssignableFrom(DimmerItem.class)) {
+                return PercentType.valueOf(transformedResponse);
+            } else {
+                return StringType.valueOf(transformedResponse);
+            }
+        } catch (Exception e) {
+            logger.debug("Couldn't create state of type '{}' for value '{}'",
+                    itemType, transformedResponse);
+            return StringType.valueOf(transformedResponse);
+        }
+    }   
+    
+	
+	
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	protected void internalReceiveCommand(String itemName, Command command) {
+		// the code being executed when a command was sent on the openHAB
+		// event bus goes here. This method is only called if one of the 
+		// BindingProviders provide a binding for the given 'itemName'.
+		logger.debug("internalReceiveCommand({},{}) is called!", itemName, command);
+		
+		/*
+		 * do we have a binding for this item at all?
+		 */
+		if (providesBindingFor(itemName)) {
+			/*
+			 * go through all the providers and look for a BindingConfig
+			 */
+			for (SallegraBindingProvider provider : providers) {
+				SallegraBindingConfig bindingConfig = provider.getBindingConfigFor(itemName);
+				
+				if (bindingConfig == null) {
```

Comment:
instead of (bindingConfig == null) continue; you could write if (bindingConfig != null) { your-business-code-here }.

At least for me, this would be more readable :smile: